core/src/test/java/io/grpc/internal/MessageFramerTest.java

```diff
@@ -323,6 +323,15 @@ public void dontCompressIfNotRequested() {
     checkStats(1000, 1000);
   }
 
+  @Test
+  public void writeRawNoopIfDisposed() {
+    byte[] bytes = new byte[]{3, 14};
+    framer.dispose();
+    framer.writeRaw(bytes, 0, bytes.length);
```

Comment:
Note that since the public method is going to check for being closed you'll need to start the write, have the bufferAllocator wait, call dispose(), wake up the bufferAllocator and check that it stops the next time around.